File: p001-p099/leetcode065.py
class Solution(object):
	def isNumber(self, s):
		"""
		:type s: str
		:rtype: bool
		"""
		hoge = re.compile("^[\ ]*[+-]?(\d+\.?\d*|\d*\.?\d+)(e[+-]?\d+)?[\ ]*$")
		return bool(hoge.match(s))
        

File: ruby/leetcode065.rb
# @param {String} s
# @return {Boolean}
def is_number(s)
    /\s*[+-]?(\d+\.\d+|\d+\.?|\.\d+)([eE][+-]?\d+)?\s*/.match(s).to_s == s and s.length > 0
end